### ISSUE TITLE
SPIKE: Investigate the best approach to test the BCGov iam-role permission boundaries

### ISSUE THREAD

**Issue by crochcunill** [bcgov/cloud-pathfinder#1591]
**Describe the issue**
One important element of the BCGov implementation of the SEA are the  iam-role permission boundaries that are put in place.

The complete set of permissions polices enforced by the different layers of system (from AWS, to GoC to BCGov) is huge. The goal in this case is to check the specific BCGov policies.

However, at this moment, it is not documented the number or roles, permissions and other constrains that are in place. 

This task will require input for other members of the team

**Background**
It seems there will be dozens if not hundreds of different policy checks. So triaging will be require otherwise the tests, even automated, will be impractically long.

They following links may contain information about the different permission layers
- https://github.com/BCDevOps/terraform-aws-keycloak-lz-sso-setup
- https://github.com/BCDevOps/terraform-aws-keycloak-lz-sso-setup/blob/a185c0cfca1299ddc5ada1551767a5291002a0f7/main.tf#L68

**Definition of done**
Basically a document(s) hopefully as short and simple as possible that will capture:
- A list of the policies that will be checked
- Triaging/prioritization of the previous list (involve stakeholders)
- Which will be the best approach to automate testing such policies (AWS CLI, API testing...)

See #1589, as probably will require long-life test accounts

**Comment by crochcunill** [comment]
Tickets #1702 and #1765 cover this issue. Actually, based on my understanding of the system, architecture and policies, this ticket is ill formed.

In addition, the first approach to testing the guardrails has been decided: comparing list of policies and roles before/after the upgrade to verify there have been no changes

In this context this ticket will be closed.
